### DISCUSSION TITLE
Is there a plan to support rules with type information?

### DISCUSSION THREAD

**Discussion by eliasm307** [angular-eslint/angular-eslint#2055] [General]
As the description says, just enquiring if there is a plan for rules to expose typescript type information in templates?

One use case I would like this for is to make an angular template version of this rule https://typescript-eslint.io/rules/no-unnecessary-type-assertion/ to highlight and possibly remove redundant non null assertions in templates. 

There are probably other type aware rules that would be useful in templates

**Comment by JamesHenry** [comment]
The Angular compiler still does not expose this information as far as I know. We would need the equivalent of the TypeScript type checker but for Angular templates to be exposed to us to leverage when writing rules.

**Reply by eliasm307** [reply]
I see, OK thanks for the explanation
